htdocs/product/class/product.class.php

```diff
@@ -1814,6 +1814,7 @@ public function getSellPrice($thirdparty_seller, $thirdparty_buyer, $pqp = 0)
 
 		// Call hook if any
 		if (is_object($hookmanager)) {
+			$hookmanager->initHooks(array('productdao'));
```

Comment:
A hook must not be intialized by classes. Hooks must be initialized in main cases by the main parent page or the main CLI script.
Module using the hook must use the context of the page.